.github/actions/setup_build/action.yml

```diff
@@ -38,8 +38,15 @@ runs:
         if [ "${{ inputs.ubuntu_version }}" = "20.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-1.3.243-focal.list https://packages.lunarg.com/vulkan/1.3.243/lunarg-vulkan-1.3.243-focal.list; fi
         if [ "${{ inputs.ubuntu_version }}" = "22.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-1.3.243-jammy.list https://packages.lunarg.com/vulkan/1.3.243/lunarg-vulkan-1.3.243-jammy.list; fi
         if [ "${{ inputs.ubuntu_version }}" = "24.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-noble.list https://packages.lunarg.com/vulkan/lunarg-vulkan-noble.list; fi
+
+        sudo wget -qO - https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --dearmor - | tee /usr/share/keyrings/llvm-archive-keyring.gpg >/dev/null
+        if [ "${{ inputs.ubuntu_version }}" = "20.04" ]; then (sudo echo 'deb [signed-by=/usr/share/keyrings/llvm-archive-keyring.gpg] http://apt.llvm.org/focal/ llvm-toolchain-focal-19 main' | sudo tee /etc/apt/sources.list.d/llvm.list >/dev/null); fi
```

Comment:
The parentheses here are confusing, they are not just for grouping like in other languages, they are specifically for creating subshells (shell environments that operate in an isolated fashion, so that e.g. variable assignments in the subshell do not last beyond it), but you don't do anything that makes use of the subshell. Is there a reason that I am missing?

Reply:
it was to keep the `|` inside the if

Reply:
You don't need parentheses for that, `if a; then b | c; fi` is valid.